torch/lib/c10d/TCPStore.hpp

```diff
@@ -125,6 +125,21 @@ class TCPStoreWorkerDaemon : public BackgroundThread {
   bool callbackRegisteredData_ = false;
 };
 
+namespace detail {
```

Comment:
curious about this detail namespace. Is it because we don't want to pollute the parent c10d namespace as there might be future applications adopting the c10d extension APIs?

Reply:
`TCPServer`, `TCPClient` and `TCPCallbackClient` are implementation details of `TCPStore`. Since they have to be member variables of `TCPStore` I forward declared them in the public header file and used pimpl idiom to hide their definitions. I also deliberately put them in the `detail` namespace to clearly indicate that they are not part of the public API.